src/inabox/inabox-resources.js

```diff
@@ -0,0 +1,266 @@
+    if (!resource) {
+      return;
+    }
+    const index = this.resources_.indexOf(resource);
+    if (index !== -1) {
+      this.resources_.splice(index, 1);
+    }
+    dev().fine(TAG, 'element removed:', resource.debugid);
+  }
+
+  /** @override */
+  removeForChildWindow(unusedChildWin) {
+    // no child window in inabox
+  }
+
+  /** @override */
+  scheduleLayoutOrPreload(unusedResource) {
+    // all elements are immediately scheduled for layout after being added
+  }
+
+  /** @override */
+  schedulePass(opt_delay) {
+    return this.pass_.schedule(opt_delay);
+  }
+
+  /** @override */
+  onNextPass(callback) {
+    this.passObservable_.add(callback);
+  }
+
+  /** @override */
+  ampInitComplete() {}
+
+  /** @override */
+  updateLayoutPriority(unusedElement, unusedNewLayoutPriority) {
+    // concept of element priority does not exist in inabox
+  }
+
+  /** @override */
+  changeSize(element, newHeight, newWidth, opt_callback, opt_newMargins) {
+    const resource = Resource.forElement(element);
+    resource./*OK*/ changeSize(newHeight, newWidth, opt_newMargins);
+    this./*OK*/ schedulePass(FOUR_FRAME_DELAY);
+    if (opt_callback) {
+      opt_callback();
+    }
+  }
+
+  /** @override */
+  attemptChangeSize(element, newHeight, newWidth, opt_newMargins) {
+    this./*OK*/ changeSize(
+      element,
+      newHeight,
+      newWidth,
+      undefined,
+      opt_newMargins
+    );
+    return Promise.resolve();
+  }
+
+  /** @override */
+  expandElement(element) {
+    const resource = Resource.forElement(element);
+    resource.completeExpand();
+    this./*OK*/ schedulePass(FOUR_FRAME_DELAY);
+  }
+
+  /** @override */
+  attemptCollapse(element) {
+    this.collapseElement(element);
+    return Promise.resolve();
+  }
+
+  /** @override */
+  collapseElement(element) {
+    const resource = this.getResourceForElement(element);
+    resource.completeCollapse();
+  }
+
+  /** @override */
+  measureElement(measurer) {
+    return this.vsync_.measurePromise(measurer);
+  }
+
+  /** @override */
+  mutateElement(element, mutator) {
+    return this.measureMutateElement(element, null, mutator);
+  }
+
+  /** @override */
+  measureMutateElement(element, measurer, mutator) {
+    return this.vsync_.runPromise({
+      measure: () => {
+        if (measurer) {
+          measurer();
+        }
+      },
+      mutate: () => {
+        mutator();
+        this./*OK*/ schedulePass(FOUR_FRAME_DELAY);
+      },
+    });
+  }
+
+  /**
+   * @return {!Promise} when first pass executed.
+   */
+  whenFirstPass() {
+    return this.firstPassDone_.promise;
+  }
+
+  /**
+   * @param {!Resource} resource
+   * @private
+   */
+  buildThenSchedulePass_(resource) {
+    this.ampdoc_
+      .whenReady()
+      .then(resource.build.bind(resource))
+      .then(this.schedulePass.bind(this, FOUR_FRAME_DELAY));
+  }
+
+  /**
+   * @private
+   */
+  doPass_() {
+    dev().fine(TAG, 'doPass');
```

Comment:
Should there be any logic for hidden states?

Reply:
do you see doc hidden would cause issue here? I'm trying to eagerly load everything no matter what the doc visibility state is.

Reply:
I'm more interested in the behavior once things are loaded/functioning. Anything needs to be paused? Animations stopped? Analytics firing?